lib/logstash/inputs/beats/message_listener.rb

```diff
@@ -119,6 +122,35 @@ def flush_buffer(ctx)
       end
     end
 
+    def extract_tls_peer(hash)
+      if @input.client_authentication_metadata?
+        tls_session = ctx.channel().pipeline().get("ssl-handler").engine().getSession()
+        tls_verified = false
+
+        # throws SSLPeerUnverifiedException if unverified
+        begin
+          tls_session.getPeerCertificates()
+          tls_verified = true
+        rescue SSLPeerUnverifiedException => e
+          if input.logger.debug?
+            input.logger.warn("Failed to get peer certificates.", :exception => e.to_s)
```

Comment:
Can you move this to just a debug message, and mention that for 'peer' mode this may be expected. Something like this: (also shouldn't need to  .to_s )
```
input.logger.debug("Failed to get peer certificates. This is normal for 'peer' verification", :exception => e)
```